maven/bnd-shared-maven-lib/src/main/java/aQute/bnd/maven/lib/resolve/ResolveSupport.java

```diff
@@ -58,4 +58,14 @@ public void setAutoRunEE(Bndrun run) {
 		run.setProperty(Constants.RUNEE, EE.getBestMatch(javaVersion)
 			.getEEName());
 	}
+
+	public void setAutoRunRequires(Bndrun run, MavenProject project) {
+		if (run.getProperty(Constants.RUNREQUIRES) == null && (project.getPackaging()
+			.equals("jar")
+			|| project.getPackaging()
+				.equals("war"))) {
+			run.setProperty(Constants.RUNREQUIRES,
+				String.format("osgi.identity;filter:='(osgi.identity=%s)'", project.getArtifactId()));
```

Comment:
Isn't this assuming that the generated bundle always uses the artifactId as the BSN? While that may be the default, it can be overridden easily and can make this auto setting quite wrong.

Reply:
This is bnd's default. But yes I do agree and loading bnd-maven-plugin configuration is part of the ongoing work. Thx